### PR TITLE
Added decent coloring

### LINKED ISSUE #2711: coloring to improve visibility
**Is your feature request related to a problem? Please describe.**

It takes a bit of searching to find out how many files were reformatted and changed

**Describe the solution you'd like**

I think adding some simple coloring to the "x files reformatted, y files left unchanged" would be great, and also additionally leaving one line before "All done! ✨ 🍰 ✨" would be great too imo.

**Describe alternatives you've considered**

Adding some colons and such would help too, but I think adding some color would be better

**Additional context**
Here is an example of what I mean

image

### PR DESCRIPTION
**Pull request by VanSHOE**
### Description

Fixes #2711 
### Checklist - did you ...

- [x] Add a CHANGELOG entry if necessary?
- [x] Add / update tests if necessary?
- [x] Add new / update outdated documentation?

Please do check if this helps

### CODE REVIEW

**Review by ichard26 — CHANGES_REQUESTED** [review]
Please add a changelog entry as this is a visible user-facing change and resolve the formatting issue (that black detected on its code :p) but other than this looks great, thank you! I'd also prefer there's a blank line before the "oh no" / "all done" line to help visually separate the summary from the rest, but this by no means will block this PR. You'll have to edit the code somewhere here: https://github.com/psf/black/blob/3fafd806b30cbff5788525f050a635639d97b11c/src/black/__init__.py#L498-L502}

A simple `out()` call before the main one should do the trick :crossed_fingers:

File: src/black/__init__.py [hunk 1]
```diff
@@ -494,6 +494,7 @@ def main(
                 workers=workers,
             )
 
+    out()
     if verbose or not quiet:
         out(error_msg if report.return_code else "All done! ✨ 🍰 ✨")
```
Comment (ichard26) [inline src/black/__init__.py:499]:
```suggestion
    if verbose or not quiet:
        out()
        out(error_msg if report.return_code else "All done! ✨ 🍰 ✨")
```
Sorry for being unclear, but we should only add the newline if we're gonna be emitting the summary anyway.

**Review by ichard26 — COMMENTED** [review]
So close!

File: src/black/__init__.py (hunk 1 above)
Comment (VanSHOE) [reply src/black/__init__.py:499]: Oh I apologize! That would be an oversight from me

**Review by ichard26 — APPROVED** [review]
Assuming CI passes, I'll merge this soon, thank you very much! Feel free to share your contributing experience and feedback here: #2238

**Comment by VanSHOE** [comment]
Ok so should I change that number to PR number in change log? I referred to the issue it was solving given that some other changes referred to the issues too

**Comment by JelleZijlstra** [comment]
It should refer to the PR number.

**Comment by ichard26** [comment]
OK, there's one case where adding a newline might be considered undesirable and I'd appreciate additional eyes on:
image

This happens when none of the files are reformatted / failed to format. Fixing this properly requires a change like this:
```diff
diff --git a/src/black/__init__.py b/src/black/__init__.py
index 67122f0..b95bc12 100644
--- a/src/black/__init__.py
+++ b/src/black/__init__.py
@@ -495,7 +495,7 @@ def main(
             )
 
     if verbose or not quiet:
-        if code is None:
+        if code is None and (verbose or report.change_count or report.failure_count):
             out()
         out(error_msg if report.return_code else "All done! ✨ 🍰 ✨")
         if code is None:
```

I really do like the spacing in all other situations though, so I'm hesitant to just drop the spacing and just merge the colouring. Thoughts?

**Comment by VanSHOE** [comment]
Sorry for the delay
I'll make a change as soon as I get back to the system

**Review by JelleZijlstra — APPROVED** [review]

**Comment by ichard26** [comment]
Thank you so much for your contribution! This project is only possible by contributions like these :black_heart:.  You're awesome, @VanSHOE. It's pretty funny though as this PR introduces colour to a project quite literally named "Black" :slightly_smiling_face: Anyway if you got any feedback to share on your contributing experience here, please lemme me know in #2238!

**Pull request merged**